### DISCUSSION TITLE
Allow containers/plugins to generate outputs

### DISCUSSION THREAD

**Discussion by qwerty287** [woodpecker-ci/woodpecker#2291] [Request a Feature]
### Clear and concise description of the problem

The only way for a plugin or another container to communicate to other containers or with the build process is to create files in the repository, which is a bad idea. 

### Suggested solution

A way for plugins or even containers to generate outputs which can be accessed from other containers or the users. Similar to GitHub Actions: https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs

### Additional context

depends on  #2954

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by 6543** [comment]
The only other generic way beside created files would be to expose an api via some socket (unix socet or udp/tcp ...)

And indicate that via some env var

**Reply by qwerty287** [reply]
I also think that's the best and most clean solution. Creating files could be easier to implement though (both woodpecker and plugin side)

**Reply by 6543** [reply]
->  #2954

**Comment by 6543** [comment]
PS: as github actions are mentioned here, i like the current architecture that divides pipelime frontemd from backend. Do validations on steps after a step did run will break this.

**Comment by 6543** [comment]
A backchannel also would be usefull for #2282 and so on.

**Comment by 6543** [comment]
Note: we should generate some sort of temporar tocken on the agent for each workflow, so you can make sure to not let other workflows malicious or unintendet interfear with things

**Reply by 6543** [reply]
(Or if falsly exposed externaly)
